src/components/GlobalPreferences/Notifications/Notifications.js

```diff
@@ -1,24 +1,232 @@
+    localStorage.getItem(NOTIFICATION_SERVICE_TOKEN_KEY)
   )
+
+  const handleAuthenticate = useCallback(() => {
+    setAuthState(AUTH_AUTHENTICATING)
+    isAuthTokenValid(token)
+      .then(v => {
+        setAuthState(AUTH_AUTHENTICATED)
+        return v
+      })
+      .catch(e => {
+        if (e instanceof TypeError) {
+          // network/service error
+          setAuthState(AUTH_SERVICE_UNAVAILABLE)
+        } else {
+          setAuthState(AUTH_AUTHENTICATION_FAILED)
+        }
+      })
+  }, [setAuthState, token])
+
+  useEffect(() => {
+    if (!email && !token) {
+      setAuthState(AUTH_UNAUTHENTICATED)
+      return
+    }
+
+    if (email && !token) {
+      setAuthState(AUTH_PREVERIFY)
+      return
+    }
+
+    if (email && token) {
+      handleAuthenticate()
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [email, token])
+
+  useEffect(() => {
+    token
+      ? localStorage.setItem(NOTIFICATION_SERVICE_TOKEN_KEY, token)
+      : localStorage.removeItem(NOTIFICATION_SERVICE_TOKEN_KEY)
+  }, [token])
+
+  useEffect(() => {
+    email
+      ? localStorage.setItem(NOTIFICATION_SERVICE_EMAIL_KEY, email)
+      : localStorage.removeItem(NOTIFICATION_SERVICE_EMAIL_KEY)
+  }, [email])
+
+  const handleLogout = useCallback(() => {
+    setEmail(null)
+    setToken(null)
+  }, [setToken, setEmail])
+
+  const setServiceUnavailable = useCallback(() => {
+    setAuthState(AUTH_SERVICE_UNAVAILABLE)
+  }, [setAuthState])
+
+  return {
+    authState,
+    email,
+    token,
+    handleTokenChange: setToken,
+    handleEmailChange: setEmail,
+    handleLogout,
+    setServiceUnavailable,
+    handleAuthenticate,
+  }
 }
 
-export default Notifications
+export default function Notifications({
+  apps,
+  subsection,
+  dao,
+  handleNavigation,
+  navigationIndex,
+}) {
+  const {
+    authState,
+    email,
+    token,
+    handleAuthenticate,
+    handleTokenChange,
+    handleEmailChange,
+    handleLogout,
+    setServiceUnavailable,
+  } = useAuthState()
+
+  const navigateToNotifications = useCallback(
+    () => handleNavigation(navigationIndex),
+    [handleNavigation, navigationIndex]
+  )
+
+  if (subsection && subsection.startsWith(VERIFY_SUBSECTION)) {
+    return (
+      <NotificationsVerify
+        subsection={subsection}
+        onTokenChange={handleTokenChange}
+        onEmailChange={handleEmailChange}
+        navigateToNotifications={navigateToNotifications}
+      />
+    )
+  }
+
+  switch (authState) {
+    case AUTH_PREVERIFY:
+      return (
+        <NotificationsPreVerify
+          email={email}
+          onEmailChange={handleEmailChange}
+        />
+      )
+    case AUTH_AUTHENTICATED:
+      return (
+        <ManageNotifications
+          onServiceUnavailable={setServiceUnavailable}
+          dao={dao}
+          onLogout={handleLogout}
+          apps={apps}
+          email={email}
+          token={token}
+        />
+      )
+    case AUTH_AUTHENTICATING:
```

Comment:
Maybe add a loader? If `AUTHENTICATING` takes some time (I recall you saying it won't but maybe we could double check this with very slow connections) then the user might be thrown off as nothing is rendered to the screen: 
![image](https://user-images.githubusercontent.com/3072458/64102439-476f0480-cd70-11e9-90e9-4c239f9100f4.png)

Reply:
Initially I had a loading ring. I removed it as it would cause a lot of flicker. 

https://github.com/aragon/aragon/commit/a1bfb22bfdbaf29ca1866f94c972b49dd68733ed

Reply:
We had a loader screen before but I remember we removed it because the process was immediate. If we are likely to encounter a blank page, I suggest we use the same feedback module, changing the title to 'Verifying your email...' and changing the icon to a loader. See below: 

<img width="1534" alt="Global Preferences - Notifications" src="https://user-images.githubusercontent.com/7205369/64109157-0500f400-cd7f-11e9-98a2-8ef1363efe10.png">

Added this screen to the Figma file.

Reply:
For the sake of clarity: 
- The comment here is about the authenticating state when the user loads the notifications ui after having successfully logged in. It is to confirm that user still exists and is valid.
- This takes around `58ms` for the server to handle and around `200ms` for the client to get the response. 
- I think the loading state for the logged-in authentication is not necessary. 
- For the verification loading state I'll update to reflect the new design

Reply:
Thank you for clarifying.

> It is to confirm that user still exists and is valid.

Is this a request to the server? If it is, my suggestion is to have an intermediate state that renders a loading ring, that's all - not every user out there might have a broadband connection as this could take more than the time you found.

Reply:
> Is this a request to the server? 

Yes. 
I'll return the loading ring but the apparent flickering will still be a problem if rendered for a short period.

Reply:
> I'll return the loading ring but the apparent flickering will still be a problem if rendered for a short period.

I understand, maybe we could give the loading some dimensions (`min-height`) that can help out with the flicker.